### PR TITLE
Bump docker version to torch 2.5.1

### PR DESCRIPTION
**Pull request by j316chuck**
Bump docker version to torch 2.5.1

### CODE REVIEW

**Comment by KuuCi** [comment]
Added your pr here:
https://github.com/mosaicml/llm-foundry/pull/1666

File: README.md
Comment (dakinggg) [inline README.md:1]: update readme should be next pr

File: .github/workflows/release.yaml
Comment (dakinggg) [inline .github/workflows/release.yaml:1]: update ci and release should be next pr

File: .github/workflows/docker.yaml [hunk 1]
```diff
@@ -17,11 +17,11 @@ jobs:
     strategy:
       matrix:
         include:
-        - name: "2.4.0_cu124"
-          base_image: mosaicml/pytorch:2.4.0_cu124-python3.11-ubuntu20.04
+        - name: "2.5.1_cu124"
```
Comment (dakinggg) [inline .github/workflows/docker.yaml:20]: sorry, one more. for this pr you should just _add_ the new docker builds, not replace the old ones

Comment (j316chuck) [reply .github/workflows/docker.yaml:26]: fixed

**Review by dakinggg — COMMENTED** [review]
will approve once the images build

**Review by dakinggg — APPROVED** [review]

**Pull request merged**